modules/assim.sequential/R/sda.enkf_refactored.R

```diff
@@ -225,32 +257,67 @@ sda.enkf <- function(settings, obs.mean, obs.cov, Q = NULL, restart=F,
                                                          stop.time = obs.times[t], 
                                                          settings = settings, 
                                                          var.names = var.names, 
-                                                         params = new.params[[i]]
-                                                         )
-                            )
+                                                         params = new.params[[i]]))
 
       # states will be in X, but we also want to carry some deterministic relationships to write_restart
       # these will be stored in params
       X[[i]]      <- X_tmp[[i]]$X
       if (!is.null(X_tmp[[i]]$params)) new.params[[i]] <- X_tmp[[i]]$params
+    }
+      X <- do.call(rbind, X)
+      FORECAST[[t]] <- X
+      
+    }else{
```

Comment:
I still think If your planning to restart a whole new sda with the outputs of an old sda you can use block of code in line 144. It already does most of the code in this `else` block.